build/sslstream-scenarios.yml

```
@@ -0,0 +1,88 @@
+# SslStream scenarios
+
+parameters:
+  - name: arguments
+    type: string
+    default: ""
+  - name: connection
+    type: string
+    default: ""
+  - name: condition
+    type: string
+    default: "true"
+
+  - name: readWriteScenarios
+    type: object
+    default:
+      - displayName: "read-write"
+        arguments: --scenario read-write --client.options.collectCounters true
+
+  - name: handshakeScenarios
+    type: object
+    default:
+      - displayName: "read-write"
+        arguments: --scenario handshake --client.options.collectCounters true
+
+  - name: tlsResume
+    type: object
+    default:
+      - displayName: "TLS Resume"
+        arguments: --variable allowTlsResume=true --property tlsResume=resume
+      - displayName: "No TLS Resume"
+        arguments: --variable allowTlsResume=false --property tlsResume=no-resume
+
+  - name: tlsVersion
+    type: object
+    default:
+      - displayName: "TLS 1.3"
+        arguments: --variable tlsVersion=1.3 --property tlsVersion=1.3
+      - displayName: "TLS 1.2"
+        arguments: --variable tlsVersion=1.2 --property tlsVersion=1.2
+
+  - name: concurrency
+    type: object
+    default:
+      - displayName: "1 connection"
+        arguments: --variable concurrency=1 --property concurrency=1
+      - displayName: "16 connections"
+        arguments: --variable concurrency=16 --property concurrency=16
+
+steps:
+  # handshake scenarios
+  - ${{ each s in parameters.handshakeScenarios }}:
+      - ${{ each version in parameters.tlsVersion }}:
+          - ${{ each res in parameters.tlsResume }}:
+              - ${{ each c in parameters.concurrency }}:
+                  - task: PublishToAzureServiceBus@1
+                    condition: succeededOrFailed()
+                    timeoutInMinutes: 10
+                    displayName: ${{ version.displayName }} ${{ s.displayName }} ${{ res.displayName }}
+                    inputs:
+                      connectedServiceName: ${{ parameters.connection }}
+                      waitForCompletion: true
+                      useDataContractSerializer: "false"
+                      messageBody: |
+                        {
+                          "condition": "(${{ parameters.condition }}) && ((new Date().getUTCHours() - 7 + 24) % 24 >= 13)",

```

```suggestion
                          "condition": "(${{ parameters.condition }}) && (Math.round(Date.now() / 43200000) % 2 == 0 )",
```

This is another way to express once every 2 half-days which I use now. Another example of how it's used: https://github.com/aspnet/Benchmarks/blob/main/build/containers-scenarios.yml#L53